src/Sylius/Behat/Context/Domain/ProductContext.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Domain;
+
+use Behat\Behat\Context\Context;
+use Sylius\Component\Core\Repository\ProductVariantRepositoryInterface;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+final class ProductContext implements Context
+{
+    /**
+     * @var RepositoryInterface
+     */
+    private $productRepository;
+
+    /**
+     * @var ProductVariantRepositoryInterface
+     */
+    private $variantRepository;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $reviewRepository;
+
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @param SharedStorageInterface $sharedStorage
+     * @param RepositoryInterface $productRepository
+     * @param ProductVariantRepositoryInterface $variantRepository
+     * @param RepositoryInterface $reviewRepository
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        RepositoryInterface $productRepository,
+        ProductVariantRepositoryInterface $variantRepository,
+        RepositoryInterface $reviewRepository
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->productRepository = $productRepository;
+        $this->variantRepository = $variantRepository;
+        $this->reviewRepository = $reviewRepository;
+    }
+
+    /**
+     * @When I delete the :productName product
```

Comment:
``:productName product`` could be handled with transformer, which would allow to remove first 6 lines from this method